welcome.php

```diff
@@ -0,0 +1,81 @@
+<?php 
+include('server.php');
+  session_start(); 
+
+  if (!isset($_SESSION['username'])) {
+  	$_SESSION['msg'] = "You must log in first";
+  	header('location: login.php');
```

Comment:
Use die() after you redirect the user. Otherwise the rest of the page still executes and can be used to exploit the server